keanu-python/keanu/algorithm/sampling.py

```diff
@@ -158,4 +177,34 @@ def _samples_generator(sample_iterator: JavaObject, vertices_unwrapped: JavaList
                 x0 += refresh_every
                 traces = []
 
-        yield sample
+        if all_scalar:
+            yield sample
+        else:
+            yield _create_multi_indexed_samples(sample, True)
+
+
+def _create_multi_indexed_samples(samples: dict, generated: bool) -> dict:
+    vertex_samples_multi: dict = {}
+    column_header_for_scalar = '(0)'
+    for vertex_label in samples:
+        vertex_samples_multi[vertex_label] = defaultdict(list)
```

Comment:
The default dict is essentially saying "I'm going to create a dictionary at this key". The contents of that dictionary are actually pretty complicated as `_add_sample_to_dict` assigns a new key for every permutation of the shape. Because it's doing the many permutations I can't have it return a simple value.

Reply:
Yep my bad, you can't create the dict entry in `__add_sample_to_dict`

But you can at least pass in one fewer parameters:

    __add_sample_to_dict(sample, vertex_samples_multi[vertex_label])